nixops/parallel.py

```diff
@@ -40,9 +40,9 @@ def print_all_backtraces(self):
 
 def run_tasks(
     nr_workers: int, tasks: Iterable[Task], worker_fun: Callable[[Task], Result]
-) -> FinalResult:
+) -> FinalResult[Result]:
     task_queue: queue.Queue[Task] = queue.Queue()
-    result_queue: queue.Queue[WorkerResult] = queue.Queue()
+    result_queue: queue.Queue[WorkerResult[Result]] = queue.Queue()
```

Comment:
I'm a bit surprised -- is FinalResult / WorkerResult a generic type?

Reply:
WorkerResult is defined a little further up, it's basically an Either from Haskell, or std::variant from C++.